src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.microsoft.azure.servicebus.MessageBody;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.exceptions.InvalidMessageException;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.model.PaymentMessage;
+
+import java.io.IOException;
+import java.util.List;
+
+public class PaymentMessageParser {
```

Comment:
should utility class be final?

Reply:
(Andrey's original point) I don't think it should be a utility class at all. If it was a normal instance class, you could inject (common) ObjectMapper to it and very importantly, you could mock this class in tests, as it will surely be some other class's dependency.

Reply:
There are controversial responses to this. Outside of this PR thought and current presentation of the class

Reply:
Why is it outside of this PR? This PR adds this class.